aws-iot-softwarepackageversion/src/main/java/software/amazon/iot/softwarepackageversion/DeleteHandler.java

```diff
@@ -0,0 +1,109 @@
+package software.amazon.iot.softwarepackageversion;
+
+import software.amazon.awssdk.http.HttpStatusCode;
+import software.amazon.awssdk.services.iot.IotClient;
+import software.amazon.awssdk.services.iot.model.DeletePackageVersionRequest;
+import software.amazon.awssdk.services.iot.model.DeletePackageVersionResponse;
+import software.amazon.awssdk.services.iot.model.GetPackageVersionRequest;
+import software.amazon.awssdk.services.iot.model.InvalidRequestException;
+import software.amazon.awssdk.services.iot.model.IotException;
+import software.amazon.awssdk.utils.StringUtils;
+import software.amazon.cloudformation.exceptions.CfnNotFoundException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.OperationStatus;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+/**
+ * The handler deletes the PackageVersion resource (if it exists)
+ * API Calls for DeleteHandler:
+ * DeletePackageVersion: To delete a PackageVersion if exists, return not found status otherwise
+ */
+public class DeleteHandler extends BaseHandlerStd {
+
+    private static final String OPERATION = "DeleteSoftwarePackageVersion";
+    private static final String CALL_GRAPH = "AWS-IoT-SoftwarePackageVersion::Delete";
+    private Logger logger;
+    private String clientToken;
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<IotClient> proxyClient,
+            final Logger logger) {
+
+        this.logger = logger;
+        this.clientToken = request.getClientRequestToken();
+
+        final ResourceModel resourceModel = request.getDesiredResourceState();
+
+        if (StringUtils.isEmpty(resourceModel.getPackageName()) || StringUtils.isEmpty(resourceModel.getVersionName())) {
```

Comment:
Why dont we do this sanity check for create handler

Reply:
some delete APIs may return success if the identifier is empty. But usually for create APIs it is universally accepted to return invalid request exception if this is not provided, hence why we generate a random string for them in the create handler.